File: src/Infectologista.ts
import { Especialista } from "./Especialista";
import { DadosSintomas } from "./DadosSintomas";

export class Infectologista extends Especialista {
  protected podeDiagnosticar(dados: DadosSintomas): boolean {
    return dados.sintomas.some(s =>
      ["febre", "calafrios", "dores no corpo"].includes(s.toLowerCase())
    );
  }

  protected diagnosticar(dados: DadosSintomas): string {
    return "Diagnóstico pelo Infectologista: possível infecção.";
  }
}

File: SistemaDiagnostico/Patterns/src/Especialistas/Infectologista.java
package Especialistas;

import Diagnostico.DadosSintomas;
import java.util.Arrays;

public class Infectologista extends Especialista {
    @Override
    protected boolean podeDiagnosticar(DadosSintomas dados) {
        return dados.getSintomas().stream()
            .map(String::toLowerCase)
            .anyMatch(s -> Arrays.asList("febre", "calafrios", "dores no corpo").contains(s));
    }

    @Override
    protected String diagnosticar(DadosSintomas dados) {
        return "Diagnóstico pelo Infectologista: possível infecção.";
    }
}
